### ISSUE TITLE
Remove architecture string from OS X builds

### ISSUE THREAD

**Issue by xycaleth** [JACoders/OpenJK#681]
OS X supports "Universal Binaries", where multiple executables of differing architectures can be packed into the same binary. OS X then chooses which binary to run based on the currently running system.

This causes problems for the builds with the recent changes in #666 because it might be that the machine that was compiling the code is x86_64, but then we choose i386 as the architecture to build for. The binary names take on the x86_64 strings, but the compiled code is actually i386.

I propose that for OS X builds, the architecture strings are not included in the binary names.

**Comment by smcv** [comment]
Do the executables really need the architecture in the name on Windows or Linux either, or is it just there because that's what the Quake III engine has traditionally done? For Debian I install them with undecorated names, like `/usr/games/openjk_sp`; you can only have one architecture installed at a time, but it doesn't seem particularly useful to have more than one anyway, since they hopefully all work identically.

If you do want architecture-flagged executables in a snapshot zip file or whatever, you already need some automation to build for all the architectures and make the zip file; so perhaps the normal build system could produce executables without the suffix, and the automation responsible for the multiple builds could rename the executables while combining them?

(It still seems advantageous to "decorate" the names for the game, cgame, ui and renderer plugins with their architecture, so a mod can have various platforms' plugins in one archive intended to be unzipped into the gamedata directory.)

**Comment by xycaleth** [comment]
Now you mention it, there doesn't seem to much benefit from adding the architecture to the executable names. We're doing it now more because that's how Quake 3 does it. @ensiform can probably comment more on this.

And yes, decorating the game/cgame/ui/renderer modules still makes sense, but not for OS X where the architectures can all be packed into the same binary.

**Comment by smcv** [comment]
Is there any advantage in *not* decorating module names on OS X? The more differences between platforms there are, the more things cannot be assumed to work on all platforms just because they work on mine, which makes development and review harder. It would seem simpler to have all platforms consistently apply these rules, at least by default:

* modules are always decorated (e.g. cgamex86.dll on Windows, uix86_64.so on Linux)
* executable names are never decorated (e.g. openjk.exe, openjk_sp on Linux)

One important factor is that the C++ code in the engine needs to know whether module names are decorated or not, whereas as far as I can see, only shortcuts/launchers will ever care what the name of the executable is.

**Comment by Razish** [comment]
Is it true that Linux can not load i386 `.so`s from x86_64?
That's a strong reason to decorate the executables - some mods are only available on certain architectures/platforms (JA+ has no 64bit build, you would need another (32 bit) executable to load it)

**Comment by xycaleth** [comment]
> Is there any advantage in not decorating module names on OS X?

I'm not suggesting this is done to gain an advantage. It's to make things work. As I previously mentioned, OS X supports Universal Binaries. A single binary, be it a .dylib file or an executable file, can contain the executable code for multiple architectures. This is *not* the same as an App Bundle, which is merely a way of packaging related resources together.

As the executable could be compiled for multiple architectures as well (we don't do this on our build servers, but someone might), it doesn't make sense to search for a .dylib file of a specific architecture.

> Is it true that Linux can not load i386 .sos from x86_64?
That's a strong reason to decorate the executables - some mods are only available on certain architectures/platforms (JA+ has no 64bit build, you would need another (32 bit) executable to load it)

Yes that's true, it's the same on Windows as well. I don't see how not decorating the executable is a problem here though. People can still *build* for x86 or x86_64, but both executables would be called `openjk.exe` for example.

**Comment by smcv** [comment]
> Is it true that Linux can not load i386 .sos from x86_64?

Yes, they are entirely separate ABIs. An x86_64 kernel can run i386 or x86_64 executables, but an x86_64 executable cannot load i386 libraries.
